src/PythonClient/src/quixstreams/dataframes/sdf/pipeline.py

```diff
@@ -0,0 +1,186 @@
+import uuid
+from typing import Self, Optional, Any, Callable, TypeAlias, List
+import operator
+from copy import deepcopy
+from quixstreams.dataframes.models import Row
+
+UNITS = {"s": "seconds", "m": "minutes", "h": "hours", "d": "days", "w": "weeks"}
+
+OpValue: TypeAlias = int | float | bool
```

Comment:
The `|` syntax is not supported in Python 3.8 yet, and Python 3.8 is our minimal supported version for now.
The pipe syntax is added in Python 3.9